### ISSUE TITLE
[OpenGL] ClearDepthStencil doesn't reset stencil mask, leading to incomplete clearing

### ISSUE THREAD

**Issue by RaZeR-RBI** [mellinoe/veldrid#448]
**Version**: 4.8.0, judging by code it's should be still present in master
**Backend**: OpenGL
**OS**: Debian 11 (should not matter though)

**Description**: OpenGL supports selective clearing of stencil buffer by using the [front stencil writemask to clear only specific bits if desired](https://stackoverflow.com/a/27415403).

However, since there's no direct way in Veldrid to configure a stencil mask without issuing a drawing operation (if I understand correctly), and [ClearDepthStencil doesn't set it](https://github.com/mellinoe/veldrid/blob/416d633c881d8b74bf01d8c174181e9943c28ccc/src/Veldrid/OpenGL/OpenGLCommandExecutor.cs#L95), stencil buffer clearing is performed only for those bits which were marked as write-enabled in the latest drawcall which used the stencil testing functionality.

Here's a sample capture from RenderDoc (I've filtered out unrelated calls for clarity):

![Screenshot](https://i.imgur.com/Ygxbfg6.png)

Here's in the latest related drawcall write mask is set to 128, and after that comes glClearStencil call, which clears only the highest bit (according to that mask), leaving data in other bits untouched.

Previous calls with mask 15 are performing increments and decrements with wrapping, and because the data in those bits isn't cleared between the frames I'm experiencing a weird effect of pixels (seemingly) randomly passing and failing stencil tests in an obscure pattern.

In short, a call to glStencilMask(~0) before glClearStencil should help, since we don't have a way to set a specific mask from the API. If anyone needs a selective clearing, one can draw a fullscreen quad with corresponding stencil test settings to override stencil values.

**Comment by mellinoe** [comment]
Thanks for the report, I applied the proposed fix (reset the stencil mask before calling `glClearStencil`), which should be fine as there's no other way to manipulate the stencil mask outside of draw operations.
